### PR TITLE
infra(server)!: fix typeorm asset entity relations

### PR DESCRIPTION
**Pull request by zackpollard**
BREAKING CHANGES 🔴
- APIs changes on the mobile app and the server, client and server must be on the same version to work.

Fixes the missing relationships in the typeorm asset entity.

### CODE REVIEW

File: server/libs/domain/src/asset/response-dto/asset-response.dto.ts [hunk 1]
```diff
@@ -32,13 +31,12 @@ export function mapAsset(entity: AssetEntity): AssetResponseDto {
...
     id: entity.id,
     deviceAssetId: entity.deviceAssetId,
-    ownerId: entity.userId,
+    ownerId: entity.ownerId,
     deviceId: entity.deviceId,
     type: entity.type,
     originalPath: entity.originalPath,
     resizePath: entity.resizePath,
     createdAt: entity.createdAt,
-    modifiedAt: entity.modifiedAt,
```
Comment (alextran1502) [inline server/libs/domain/src/asset/response-dto/asset-response.dto.ts:41]: This is the file-level modified time. Perhaps you think it was the database-related modified time?

**Comment by alextran1502** [comment]
For the completeness of this PR, I think we have to regenerate the OpenAPI and modify the changes on the mobile client and the web client.

File: server/libs/domain/src/asset/response-dto/asset-response.dto.ts (hunk 1 above)
Comment (zackpollard) [reply server/libs/domain/src/asset/response-dto/asset-response.dto.ts:41]: Yep, missed that, have changed the file related ones to fileModifiedAt and fileCreatedAt, and have also added createdAt to go alongside updatedAt like we have for most entities.

File: server/apps/immich/src/api-v1/asset/asset.core.ts [hunk 2]
```diff
@@ -19,24 +18,23 @@ export class AssetCore {
...
 
+      fileCreatedAt: dto.createdAt,
+      fileModifiedAt: dto.modifiedAt,
+
       type: dto.assetType,
       isFavorite: dto.isFavorite,
       duration: dto.duration || null,
       isVisible: dto.isVisible ?? true,
-      livePhotoVideoId: livePhotoAssetId || null,
+      livePhotoVideo: livePhotoAssetId != null ? ({ id: livePhotoAssetId } as AssetEntity) : null,
```
Comment (jrasm91) [inline server/apps/immich/src/api-v1/asset/asset.core.ts:37]: Why this change here? We still have the `livePhotoVideoId` field, right?

File: web/src/lib/utils/file-uploader.ts [hunk 3]
```diff
@@ -83,10 +83,10 @@ async function fileUploader(
 		formData.append('assetType', assetType);
 
 		// Get Asset Created Date
-		formData.append('createdAt', createdAt);
+		formData.append('fileCreatedAt', fileCreatedAt);
 
 		// Get Asset Modified At
-		formData.append('modifiedAt', new Date(asset.lastModified).toISOString());
+		formData.append('fileModifiedAt', new Date(asset.lastModified).toISOString());
```
Comment (jrasm91) [inline web/src/lib/utils/file-uploader.ts:89]: Looks like this is the same value as `fileCreatedAt`. :shrug:

**Review by jrasm91 — APPROVED** [review]
Not sure if you can directly cast from varchar to uuid, other than that looks good.

File: server/apps/immich/src/api-v1/asset/asset.core.ts (hunk 2 above)
Comment (zackpollard) [reply server/apps/immich/src/api-v1/asset/asset.core.ts:37]: We do, but I think we can remove those in general and use this way of writing existing assets.

File: web/src/lib/utils/file-uploader.ts (hunk 3 above)
Comment (zackpollard) [reply web/src/lib/utils/file-uploader.ts:89]: Yep, I just kept the existing functionality. Potentially we can't access the createdAt here? Not really sure why it was done this way previously.
